### PR TITLE
WIP: Feature/cli config

### PR DESCRIPTION
**Pull request by sshaman1101**
Related to #34

### CODE REVIEW

File: cmd/cli/main.go [hunk 1]
```diff
@@ -47,7 +51,30 @@ func checkHubAddressIsSet(cmd *cobra.Command, args []string) error {
...
 }
 
+// commandError allow to present any internal error as JSON
+type commandError struct {
+	Err     error  `json:"error"`
+	Message string `json:"message"`
+}
+
+func (ce *commandError) ToJSONString() string {
+	j, _ := json.Marshal(ce)
```
Comment (noxiouz) [inline cmd/cli/main.go]: https://play.golang.org/p/jfoAwV3epl

Comment (sshaman1101) [inline cmd/cli/main.go]: I haven't ever meet that behaviour.
I can use little hack with unmarshallable error field and marshallable string field, like this:
https://play.golang.org/p/FtYxrFBiJj

**Comment by sshaman1101** [comment]
@noxiouz @3Hren PTAL

File: cmd/cli/commands/common.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
+package commands
+
+import (
+	"context"
+	b64 "encoding/base64"
+	"encoding/json"
+	"errors"
+	"time"
+
```
Comment (noxiouz) [inline cmd/cli/commands/common.go]: no empty line

File: cmd/cli/commands/common.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	auth := types.AuthConfig{
+		Username:      login,
+		Password:      password,
+		ServerAddress: registry,
+	}
+	jsonAuth, _ := json.Marshal(auth)
+	return b64.StdEncoding.EncodeToString(jsonAuth)
+}
+
+func checkHubAddressIsSet(cmd *cobra.Command, args []string) error {
```
Comment (noxiouz) [inline cmd/cli/commands/common.go]: unused variable args

File: cmd/cli/commands/common.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+func newCommandError(message string, err error) *commandError {
+	return &commandError{rawErr: err, Message: message}
+}
+
+func showError(message string, err error) {
+	if cfg.OutputFormat() == config.OutputModeSimple {
+		showErrorInSimple(message, err)
+	} else {
+		showErrorInJSON(message, err)
+
```
Comment (noxiouz) [inline cmd/cli/commands/common.go]: empty line

**Comment by noxiouz** [comment]
Why do not u use https://godoc.org/github.com/spf13/cobra#Command.OutOrStderr + fmt.Fprintf instead of fmt.Printf

**Pull request merged**